Code/GraphMol/Descriptors/BagOfBonds.cpp

```diff
@@ -0,0 +1,310 @@
+    if (res>0) {
+      std::pair<std::string,unsigned int > key = std::make_pair(PairString, res);
+      ks.push_back(key);
+    }
+  }
+  return ks;
+
+}
+
+
+ BagMatrix getCoulombMatBags(Eigen::VectorXd numbers, int numatoms, Eigen::MatrixXd Distance3D,  std::vector<std::string > S, int alpha) {
+  // 3D distance matrix
+  Eigen::MatrixXd ProdV = numbers*numbers.transpose(); // outer products of vector ie (z[i] * z[j])
+
+  if (alpha != 1) {
+    Distance3D = Distance3D.array().pow(alpha); // power distance alpha
+  }
+
+  Eigen::MatrixXd MyVal = ProdV.cwiseQuotient(Distance3D); // ratio top / dist
+  MyVal.diagonal() = 0.5 * numbers.array().pow(2.4); // set the diagonal fix values 
+  // get the BagsCode
+  std::vector<std::string > BagsCodes;
+  std::list<std::pair<std::string, double > > Res;
+  std::string s1, s2, key;
+  for (unsigned int i = 0; i< numatoms; i++) {
+    s1 = S[i];
+    BagsCodes.push_back(s1); // adding the diagonal once!
+
+    for (unsigned int j = i+1; j< numatoms; j++ ){
+      s2 = S[j];
+      if (s1 > s2) {
+        key = s2+s1;
+      }
+      else {
+        key = s1+s2;
+      }
+      BagsCodes.push_back(key);
+    }
+  }    
+
+  Eigen::VectorXd RES = getUpperMat(MyVal);
+  return BagMatrix{RES , BagsCodes}; 
+}
+
+///// code to be expose to python directly not simple
+// we can parallelized cause we need the max value per key
+ std::map<std::string, unsigned int> getBoBSizefromSmiles(std::vector<std::string> smiles) {
+    //unsigned int idx = 0;
+    //std::vector< std::list < std::pair<std::string, unsigned int> > > myData(smiles.size());
+    std::list<std::pair<std::string, unsigned int> > fulldata;
+    std::map<std::string, unsigned int> Global;
+    std::string key;
+
+    //while (idx < smiles.size()) {
+    for (auto const& smi :  smiles) {
+      RDKit::ROMol *mol;
+      mol = RDKit::SmilesToMol(smi);
+      std::list<std::pair<std::string, unsigned int>> data = EnumeratesAtomsPair(*mol, false, false);
+
+      for (auto const& pair :  data) {
+          key = pair.first;
+          // don't need to find the value just if exists! count instead of find is faster
+          if (Global.count(key) > 0) {
+            if (pair.second > Global[key]) {
+              Global[key] = pair.second;
+            }   
+          }         
+          else {
+              Global[key] = pair.second;
+          }
+      }
+
+      //++idx;
+      delete mol;
+    }
+
+
+    return Global;
+}
+
+void getBagOfBonds(const RDKit::ROMol &mol, std::vector<double> &res, double *dist3D, 
+unsigned int numAtoms, std::map<std::string, unsigned int> MaxBags, int alpha) {
+
+  // initialize the variables for the getCoulombMat
+  int numatoms= mol.getNumAtoms();
+  double *z = new double[numatoms];
+  std::vector<std::string > S;
+  for (int i=0; i< numatoms; i++){
+      z[i] = mol.getAtomWithIdx(i)->getAtomicNum();
+      S.push_back(mol.getAtomWithIdx(i)->getSymbol());
```

Comment:
In `EnumerateAtomsPair()` you use the symbol from the periodic table instead of the symbol from the atom (which can be set by the user), I think this is a better idea.

Reply:
I don't see what/where should I change then ?

Reply:
Yeah, I think it's a better idea to use the symbol from the periodic table here.

Reply:
done